test/models/driver_test.rb

```diff
@@ -58,24 +58,72 @@
     end
   end
 
-  # Tests for methods you create should go here
   describe "custom methods" do
+    before do
+      @driver = Driver.create(name: "Kari", vin: "123", available: true)
+      @new_passenger = Passenger.create(name: "Kari", phone_num: "111-111-1211")
+      @trip_1 = Trip.create(driver_id: @driver.id, passenger_id: @new_passenger.id, date: Date.today, rating: 5, cost: 1234)
+      @trip_2 = Trip.create(driver_id: @driver.id, passenger_id: @new_passenger.id, date: Date.today, rating: 3, cost: 6334)
+    end
+
     describe "average rating" do
-      # Your code here
+      it "returns the correct average rating" do
+        expect(@driver.average_rating).must_be_close_to (5.0 + 3.0) / 2.0, 0.01
+      end
+
+     it "returns a float" do
+      expect(@driver.average_rating).must_be_kind_of Float
+     end
+    
+      it "returns zero if no driven trips" do
+        driver = Driver.new(
+          name: "Rogers Bartell IV",
+          vin: "1C9EVBRM0YBC564DZ",
+          available: false
+        )
+        expect(driver.average_rating).must_equal 0
+      end
     end
+  end
 
-    describe "total earnings" do
-      # Your code here
+  describe "total earnings" do
+    before do
+      @driver = Driver.create(name: "Kari", vin: "123", available: true)
+      @new_passenger = Passenger.create(name: "Kari", phone_num: "111-111-1211")
+      @trip_1 = Trip.create(driver_id: @driver.id, passenger_id: @new_passenger.id, date: Date.today, rating: 5, cost: 1234)
+      @trip_2 = Trip.create(driver_id: @driver.id, passenger_id: @new_passenger.id, date: Date.today, rating: 3, cost: 6334)
     end
 
-    describe "can go online" do
-      # Your code here
+    it "returns float" do
+      expect(@driver.total_revenue).must_be_kind_of Float
```

Comment:
The tests in driver are quite thorough for the most part! This is an opportunity for improvement, where you could test the exact value of the revenue. 
```suggestion
      expect(@driver.total_revenue).must_be_kind_of Float
      expect(@driver.total_revenue).must_equal 7568
```